sklearn/tree/tests/test_tree.py

```diff
@@ -2152,7 +2160,11 @@ def test_decision_tree_regressor_sample_weight_consistentcy(criterion):
     # Thresholds, tree.tree_.threshold, and values, tree.tree_.value, are not
     # exactly the same, but on the training set, those differences do not
     # matter and thus predictions are the same.
-    assert_allclose(tree1.predict(X), tree2.predict(X))
+    y_pred1 = tree1.predict(X)
+    y_pred2 = tree2.predict(X)
+    # TODO: y_pred1 and y_pred2 will not match global_dtype for 32 bit
+    # assert y_pred1.dtype == global_dtype
+    assert_allclose(y_pred1, y_pred2)
```

Comment:
> Whilst tree-based models do not preserve dtypes, we must still test that they work on 32bit datasets -- #22881 can still be addressed for this case.

Is this what you mean @jjerphan ?

Reply:
Yes, the changes you made above `R2142-R2144` (i.e. https://github.com/scikit-learn/scikit-learn/pull/22926/files#diff-e0c534f299457278700ee68add18acf0d4b2a1dd9391e1bcc49f3fe1b472fe67R2142-R2144) is part of what is expected.

Reply:
Okay this is done now.